uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/EventStorage.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt.client.control.api;
+
+import lombok.NonNull;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Interface to storage of the events.
+ */
+public interface EventStorage {
+
+    /**
+     * Removes all events from storage.
+     */
+    void clear();
+
+    /**
+     * Adds event to storage.
+     *
+     * @param event the event to add
+     */
+    void addEvent(@NonNull Event event);
+
+    /**
+     * Gets matched events from storage immediately.
+     *
+     * @param filter the filter of events
+     * @return the array of matched events
+     */
+    Event[] getEvents(@NonNull EventFilter filter);
+
+    /**
+     * Waiting for matched events.
+     *
+     * @param filter the filter of events
+     * @param timeout the maximum time to wait
+     * @param unit the time unit of the timeout argument
+     * @return the array of matched events
+     * @throws TimeoutException if the wait timed out
+     * @throws InterruptedException if the current thread was interrupted while waiting
+     */
+    Event[] awatingEvents(@NonNull EventFilter filter, long timeout, @NonNull TimeUnit unit)
```

Comment:
It is nice to have, but for tests we can use just sleep with following getEvents call

Reply:
It really required due to you can't predict when message will be received.
Sleep will suspend test execution all the time, passing all tests will takes so long time.

Reply:
For example: you waiting for 60 seconds max. With sleep(60) is takes always at least 60 seconds.
With awatingEvents(60) it takes only 10 seconds if message will be received after 10 seconds.
We save 50 seconds on one test. With greengrass we can have thousands of tests.